### PR TITLE
Release/2.3.0

### PR DESCRIPTION
**Pull request by jbeemster**

### CODE REVIEW

File: tracker/tracker.go [hunk 1]
```diff
@@ -90,6 +94,38 @@ func (t Tracker) FlushEmitter() {
 	t.Emitter.Flush()
 }
 
+func (t *Tracker) waitForEmitter() {
+	for {
+		if !t.Emitter.IsSending() || t.Emitter.SendChannel == nil {
+			break
+		}
+		time.Sleep(1 * time.Second)
```
Comment (paulboocock) [inline tracker/tracker.go:102]: 1 second seems like quite a long time to wait. Not a huge issue for some applications (i.e. short lived ones) but if I wanted to use this tracker and BlockingFlush then I might want to check more frequently so I can move on sooner. Maybe make this configurable, specify it in milliseconds and inject the wait time in? Then if I want I could choose 100ms or something.

**Comment by paulboocock** [comment]
Oh, maybe add a test to cover some of the logic in `BlockingFlush()`?

File: tracker/tracker.go (hunk 1 above)
Comment (jbeemster) [reply tracker/tracker.go]: Refactored to make it user configurable with ability to supply a milliseconds to sleep between each flush attempt.

**Comment by jbeemster** [comment]
Fixed the coverage - going to merge it in!

**Pull request merged**